schema/deploy/shipping/views.sql

```diff
@@ -844,4 +844,73 @@ comment on view shipping.hcov19_observation_v1 is
   'Custom view of hCoV-19 samples with presence-absence results and best available encounter data';
 
 
+create or replace view shipping.scan_return_results_v1 as
+    select distinct on (barcode, lineage)
+        sample_id,
+        barcode as qrcode,
+        encountered as collect_ts,
+        case
+            when sample_id is null then 'not-received'
+            when sample_id is not null and presence_absence_id is null then 'pending'
+            when present is true then 'positive'
+            when present is false then 'negative'
+            when presence_absence_id is not null and present is null then 'inconclusive'
+        end as status_code,
+        presence_absence_id, -- todo debug
+        target.identifier,  -- todo debug
+        lineage,  -- todo debug
+        present  -- todo debug
+
+    from warehouse.identifier
+    join warehouse.identifier_set using (identifier_set_id)
+    left join warehouse.sample on uuid::text = sample.collection_identifier
+    left join warehouse.encounter using (encounter_id)
+    left join warehouse.presence_absence using (sample_id)
+    left join warehouse.target using (target_id)
+    left join warehouse.organism using (organism_id)
+
+    where identifier_set.name in('collections-scan', 'collections-swab&send') -- todo
+        and target.control is distinct from true
+        and (lineage is null or
+            lineage <@ '{"Human_coronavirus.2019"}'::ltree[])
```

Comment:
The combination of left joins across multiple tables above + the where condition `lineage is null or lineage <@ …` here suffers from the same problems as we discussed in the `hcov19_observation_v1` view. It works now, but there is the chance for future targets without a linked organism to slip into these results. That'd be bad, as the view would report a result for a non-hCoV-19 target. I think better to join to a subquery or CTE which selects just the hCoV-19 results, similar to what was done in `hcov19_observation_v1`.

Reply:
Ah, right. I pushed a new commit that addresses this bug.